interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/KotlinPoetMetadataSpecs.kt

```diff
@@ -648,34 +648,32 @@ private fun KmProperty.toPropertySpec(
   val returnTypeName = returnType.toTypeName(typeParamResolver)
   val mutableAnnotations = mutableListOf<AnnotationSpec>()
   if (containerData != null && propertyData != null) {
-    if (hasGetter) {
-      getterSignature?.let { getterSignature ->
-        if (!containerData.isInterface &&
-          modality != Modality.OPEN && modality != Modality.ABSTRACT
+    getterSignature?.let { getterSignature ->
+      if (!containerData.isInterface &&
+        modality != Modality.OPEN && modality != Modality.ABSTRACT
+      ) {
+        // Infer if JvmName was used
+        // We skip interface types or open/abstract properties because they can't have @JvmName.
+        // For annotation properties, kotlinc puts JvmName annotations by default in
+        // bytecode but they're implicit in source, so we expect the simple name for
+        // annotation types.
+        val expectedMetadataName = if (containerData is ClassData &&
+          containerData.declarationContainer.isAnnotation
         ) {
-          // Infer if JvmName was used
-          // We skip interface types or open/abstract properties because they can't have @JvmName.
-          // For annotation properties, kotlinc puts JvmName annotations by default in
-          // bytecode but they're implicit in source, so we expect the simple name for
-          // annotation types.
-          val expectedMetadataName = if (containerData is ClassData &&
-            containerData.declarationContainer.isAnnotation
-          ) {
-            name
-          } else {
-            "get${name.safeCapitalize(Locale.US)}"
-          }
-          getterSignature.jvmNameAnnotation(
-            metadataName = expectedMetadataName,
-            useSiteTarget = UseSiteTarget.GET,
-          )?.let { jvmNameAnnotation ->
-            mutableAnnotations += jvmNameAnnotation
-          }
+          name
+        } else {
+          "get${name.safeCapitalize(Locale.US)}"
+        }
+        getterSignature.jvmNameAnnotation(
+          metadataName = expectedMetadataName,
+          useSiteTarget = UseSiteTarget.GET,
+        )?.let { jvmNameAnnotation ->
+          mutableAnnotations += jvmNameAnnotation
         }
       }
     }
-    if (hasSetter) {
-      setterSignature?.let { setterSignature ->
+    if (setter != null) {
+        setterSignature?.let { setterSignature ->
```

Comment:
nit (should be caught by spotless):

```suggestion
      setterSignature?.let { setterSignature ->
```